### ISSUE TITLE
Unexpected token kind tk.OP when __all__ contains non str literals; noqa does not help

### ISSUE THREAD

**Issue by OJFord** [PyCQA/pydocstyle#313]
With an `__all__` like:
```python
from .other import __all__ as other_all
from .other import *

__all__ = (
    *other_all,
    'something',
)
```

I see:
```
WARNING: Error in file path/to/__init__.py: Unexpected token kind in  __all__: tk.OP.
That means pydocstyle cannot decide which definitions are
public. Variable __all__ should be present at most once in
each file, in form
`__all__ = ('a_public_function', 'APublicClass', ...)`.
More info on __all__: http://stackoverflow.com/q/44834/. ')
```

Thinking the 'present at most once' was a clue, I changed to using `from . import other` and `*other.__all__`, but got the same error.

Either of these seems to _work_ as Python, but it cannot be ignored with `# noqa`.

**Comment by OJFord** [comment]
@shacharoo Will it? - the error persists even _with_ a module docstring.

I would have thought that resolving #297 will just mean that if there's no docstring it complains about that; then either way still chokes on non-trivial `__all__`s.

**Comment by shacharoo** [comment]
@OJFord  - @Nurdok and I discussed it and we're gonna change the __all__ parsing to be best-effort. If the parsing of __all__ fails we will just assume that every "public looking" function / class will be checked (just like it is now: https://github.com/PyCQA/pydocstyle/blob/676e4f3100c3c789ea23996b505f093e9c455016/src/pydocstyle/parser.py#L141)

**Comment by Nurdok** [comment]
Fixed by #320.
